persistence-modules/java-mongodb/src/main/java/com/baeldung/morphia/domain/Book.java

```diff
@@ -1,116 +1,168 @@
-@Validation("{ price : { $gt : 0 } }")
-public class Book {
-    @Id
-    private String isbn;
-    @Property
-    private String title;
-    private String author;
-    @Embedded
-    private Publisher publisher;
-    @Property("price")
-    private double cost;
-    @Reference
-    private Set<Book> companionBooks;
-
-    public Book() {
-
-    }
-
-    public String getTitle() {
-        return title;
-    }
-
-    public String getAuthor() {
-        return author;
-    }
-
-    public double getCost() {
-        return cost;
-    }
-
-    public void addCompanionBooks(Book book) {
-        if (companionBooks != null)
-            this.companionBooks.add(book);
-    }
-
-    public Book(String isbn, String title, String author, double cost, Publisher publisher) {
-        this.isbn = isbn;
-        this.title = title;
-        this.author = author;
-        this.cost = cost;
-        this.publisher = publisher;
-        this.companionBooks = new HashSet<>();
-    }
-
-    @Override
-    public String toString() {
-        return "Book [isbn=" + isbn + ", title=" + title + ", author=" + author + ", publisher=" + publisher + ", cost=" + cost + "]";
-    }
-
-    @Override
-    public int hashCode() {
-        final int prime = 31;
-        int result = 1;
-        result = prime * result + ((author == null) ? 0 : author.hashCode());
-        long temp;
-        temp = Double.doubleToLongBits(cost);
-        result = prime * result + (int) (temp ^ (temp >>> 32));
-        result = prime * result + ((isbn == null) ? 0 : isbn.hashCode());
-        result = prime * result + ((publisher == null) ? 0 : publisher.hashCode());
-        result = prime * result + ((title == null) ? 0 : title.hashCode());
-        return result;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        if (this == obj)
-            return true;
-        if (obj == null)
-            return false;
-        if (getClass() != obj.getClass())
-            return false;
-        Book other = (Book) obj;
-        if (author == null) {
-            if (other.author != null)
-                return false;
-        } else if (!author.equals(other.author))
-            return false;
-        if (Double.doubleToLongBits(cost) != Double.doubleToLongBits(other.cost))
-            return false;
-        if (isbn == null) {
-            if (other.isbn != null)
-                return false;
-        } else if (!isbn.equals(other.isbn))
-            return false;
-        if (publisher == null) {
-            if (other.publisher != null)
-                return false;
-        } else if (!publisher.equals(other.publisher))
-            return false;
-        if (title == null) {
-            if (other.title != null)
-                return false;
-        } else if (!title.equals(other.title))
-            return false;
-        return true;
-    }
-
+package com.baeldung.morphia.domain;

```

Comment:
Again, can't tell what was actually changed. If it's a whitespace-only change, then we should revert the changes.